### PR TITLE
feat: migrate to preact/compat

### PR DESCRIPTION
**Pull request by SgtPooki**
## Title

feat: migrate to preact/compat

## Description

Built on top of https://github.com/ipfs-shipyard/service-worker-gateway/pull/188 intended to test migration to preact. **The size reduction is 0.5M**

## Notes & open questions

Things work fine, but preact/compat says suspense and lazy are experimental and shouldn't be used in prod.

#### Size and assets on `fix/dynamic-imported-ui-components`

```bash
total 4.7M
drwxr-xr-x 33 sgtpooki 1.1K Apr 10 13:41 ..
-rw-r--r--  1 sgtpooki  30K Apr 10 13:41 ipfs-sw-778.js
-rw-r--r--  1 sgtpooki 5.2K Apr 10 13:41 ipfs-sw-778.css
-rw-r--r--  1 sgtpooki  51K Apr 10 13:41 ipfs-sw-main.js
-rw-r--r--  1 sgtpooki  16K Apr 10 13:41 ipfs-sw-781.js
-rw-r--r--  1 sgtpooki  25K Apr 10 13:41 ipfs-sw-850.js
-rw-r--r--  1 sgtpooki 2.5K Apr 10 13:41 ipfs-sw-992.js
-rw-r--r--  1 sgtpooki 2.5K Apr 10 13:41 ipfs-sw-611.js
-rw-r--r--  1 sgtpooki  55K Apr 10 13:41 ipfs-sw-sw.js
-rw-r--r--  1 sgtpooki 2.7K Apr 10 13:41 ipfs-sw-styles.css
-rw-r--r--  1 sgtpooki  57K Apr 10 13:41 ipfs-sw-117.js
-rw-r--r--  1 sgtpooki  223 Apr 10 13:41 ipfs-sw-styles.js
-rw-r--r--  1 sgtpooki  88K Apr 10 13:41 ipfs-sw-96.js
-rw-r--r--  1 sgtpooki   26 Apr 10 13:41 _redirects
-rw-r--r--  1 sgtpooki 577K Apr 10 13:41 ipfs-sw-vendor-react.js
-rw-r--r--  1 sgtpooki 3.4M Apr 10 13:41 ipfs-sw-561.js
-rw-r--r--  1 sgtpooki 6.1K Apr 10 13:41 ipfs-sw-117.js.LICENSE.txt
drwxr-xr-x 31 sgtpooki  992 Apr 10 13:41 .
-rw-r--r--  1 sgtpooki 284K Apr 10 13:41 ipfs-sw-561.js.LICENSE.txt
-rw-r--r--  1 sgtpooki   43 Apr 10 13:41 ipfs-sw-611.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 2.8K Apr 10 13:41 ipfs-sw-781.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 1.9K Apr 10 13:41 ipfs-sw-778.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 8.3K Apr 10 13:41 ipfs-sw-96.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 1.5K Apr 10 13:41 ipfs-sw-850.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 8.3K Apr 10 13:41 ipfs-sw-sw.js.LICENSE.txt
-rw-r--r--  1 sgtpooki   43 Apr 10 13:41 ipfs-sw-992.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 5.9K Apr 10 13:41 ipfs-sw-main.js.LICENSE.txt
-rw-r--r--  1 sgtpooki    6 Apr 10 13:41 ipfs-sw-styles.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 1.7K Apr 10 13:41 index.html
-rw-r--r--  1 sgtpooki  776 Apr 10 13:41 ipfs-sw-vendor-react.js.LICENSE.txt
-rw-r--r--  1 sgtpooki  34K Apr 10 13:41 favicon.ico
```

#### Size and assets on this branch

```bash
total 4.2M
drwxr-xr-x 33 sgtpooki 1.1K Apr 10 13:37 ..
-rw-r--r--  1 sgtpooki  51K Apr 10 13:37 ipfs-sw-main.js
-rw-r--r--  1 sgtpooki  30K Apr 10 13:37 ipfs-sw-778.js
-rw-r--r--  1 sgtpooki 5.2K Apr 10 13:37 ipfs-sw-778.css
-rw-r--r--  1 sgtpooki  16K Apr 10 13:37 ipfs-sw-781.js
-rw-r--r--  1 sgtpooki  25K Apr 10 13:37 ipfs-sw-850.js
-rw-r--r--  1 sgtpooki 2.5K Apr 10 13:37 ipfs-sw-611.js
-rw-r--r--  1 sgtpooki  55K Apr 10 13:37 ipfs-sw-sw.js
-rw-r--r--  1 sgtpooki 2.5K Apr 10 13:37 ipfs-sw-992.js
-rw-r--r--  1 sgtpooki 2.7K Apr 10 13:37 ipfs-sw-styles.css
-rw-r--r--  1 sgtpooki  223 Apr 10 13:37 ipfs-sw-styles.js
-rw-r--r--  1 sgtpooki  57K Apr 10 13:37 ipfs-sw-117.js
-rw-r--r--  1 sgtpooki   26 Apr 10 13:37 _redirects
-rw-r--r--  1 sgtpooki 174K Apr 10 13:37 ipfs-sw-304.js
-rw-r--r--  1 sgtpooki 6.1K Apr 10 13:37 ipfs-sw-117.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 3.4M Apr 10 13:37 ipfs-sw-561.js
drwxr-xr-x 29 sgtpooki  928 Apr 10 13:37 .
-rw-r--r--  1 sgtpooki   43 Apr 10 13:37 ipfs-sw-611.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 2.1K Apr 10 13:37 ipfs-sw-778.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 8.5K Apr 10 13:37 ipfs-sw-304.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 284K Apr 10 13:37 ipfs-sw-561.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 1.7K Apr 10 13:37 ipfs-sw-850.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 2.9K Apr 10 13:37 ipfs-sw-781.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 6.3K Apr 10 13:37 ipfs-sw-main.js.LICENSE.txt
-rw-r--r--  1 sgtpooki    6 Apr 10 13:37 ipfs-sw-styles.js.LICENSE.txt
-rw-r--r--  1 sgtpooki  34K Apr 10 13:37 favicon.ico
-rw-r--r--  1 sgtpooki   43 Apr 10 13:37 ipfs-sw-992.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 8.3K Apr 10 13:37 ipfs-sw-sw.js.LICENSE.txt
-rw-r--r--  1 sgtpooki 1.6K Apr 10 13:37 index.html
```

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by SgtPooki** [comment]
FYI: I think the lightweight usage of suspense here should be fine for now.. we can address later if we're really concerned about Suspense/lazy being experimental. lmk if you want me to fix now, and I can do some async finagling.

File: src/pages/config.tsx
Comment (lidel) [inline src/pages/config.tsx:1]: Seems there is a small cosmetic(?) regression when I open `http://sw.localhost:{port}/#/ipfs-sw-config` and try to customize gateways and then save them. 

Pressing on <kbd>Save Config</kbd> button seems to save the new value, but it does not switch me back to `http://sw.localhost:{port}/`, like it did in the react version, I am stuck on the config screen:

> 2024-04-15-235433_1027x585_scrot

**Review by lidel — COMMENTED** [review]
We are down to ~4.2MiB uncompressed, really nice! 

I've built with `npm ci && npm run build` and tested with `cd dist && python -m http.server 3000`, found small cosmetic problem below, other than that, lgtm.

File: src/pages/config.tsx
Comment (SgtPooki) [reply src/pages/config.tsx:1]: This changed slightly recently. Ill double check

Comment (SgtPooki) [reply src/pages/config.tsx:1]: So weird. I was not able to reproduce this, then I was, then I wasn't... I will merge main and then look some more.

Comment (SgtPooki) [reply src/pages/config.tsx:1]: https://github.com/ipfs-shipyard/service-worker-gateway/assets/1173416/e9a9b7ea-9eca-4dbf-b339-896599dcfea9

Comment (SgtPooki) [reply src/pages/config.tsx:1]: you know what it was, my dev server was shut down and the redirect didn't work when it was down. it works when it stays up..

**Comment by SgtPooki** [comment]
config page save -> redirect is happening with `npx http-server dist` as well. I think it's a non-issue? we can address later if someone can replicate that problem, I can't right now!

**Pull request merged**